src/core/core.controller.js

```diff
@@ -643,12 +660,21 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 			return;
 		}
 
-		// Draw all the scales
-		helpers.each(me.boxes, function(box) {
-			box.draw(me.chartArea);
-		}, me);
+		// Because of plugin hooks (before/afterDatasetsDraw), datasets can't
```

Comment:
Why do the plugin hooks stop datasets from being part of the layers? I'm not sure I see what the issue would be

Reply:
`beforeDatasetsDraw` is expected to be called before all datasets are drawn (but after stuff drawn under datasets) and `afterDatasetsDraw` is expected to be called after all datasets are drawn (but before stuff over datasets). That means drawing that happen between these 2 hooks should not include anything else that datasets.

If datasets were layers, they would be mixed with scales, etc., thus would make these 2 hooks irrelevant and probably introduce a breaking change. Though, datasets as layers is something we also talked with @kurkle but should be investigated in another PR.